qiskit_experiments/analysis/curve_fit_analysis.py

```diff
@@ -0,0 +1,244 @@
+        bounds: Optional[Tuple] = None,
+        fit_mean_data: bool = False,
+        plot: bool = True,
+        ax: Optional["AxesSubplot"] = None,
+        **kwargs,
+    ):
+        """Run curve fit analysis on circuit data.
+
+        Args:
+            experiment_data (ExperimentData): the experiment data to analyze.
+            func: fit function `f(x, *params)`.
+            p0: Optional, initial parameter values for curve_fit.
+            p0_func: Optional, function for calculating initial p0.
+            bounds: Optional, parameter bounds for curve_fit.
+            fit_mean_data: Optional, if True pass means of data points to curve_fit.
+            plot: If True generate a plot of fitted data.
+            ax: Optional, matplotlib axis to add plot to.
+            kwargs: additional kwargs to pass to curve_fit.
+
+        Returns:
+            tuple: A pair ``(analysis_results, figures)`` where
+                   ``analysis_results`` may be a single or list of
+                   AnalysisResult objects, and ``figures`` may be
+                   None, a single figure, or a list of figures.
+        """
+        return self._run_curve_fit(
+            experiment_data.data,
+            func,
+            p0=p0,
+            p0_func=p0_func,
+            bounds=bounds,
+            fit_mean_data=fit_mean_data,
+            plot=plot,
+            ax=ax,
+            **kwargs,
+        )
+
+    def _run_curve_fit(
+        self,
+        data: List[Dict[str, any]],
+        func: Callable,
+        p0: Optional[List[float]] = None,
+        p0_func: Optional[Callable] = None,
+        bounds: Optional[Tuple] = None,
+        fit_mean_data: bool = False,
+        plot: bool = True,
+        ax: Optional["AxesSubplot"] = None,
+        **kwargs,
+    ):
+        """Run curve fit analysis on circuit data.
+
+        Args:
+            data (ExperimentData): the experiment data to analyze.
+            func: fit function `f(x, *params)`.
+            p0: Optional, initial parameter values for curve_fit.
+            p0_func: Optional, function for calculating initial p0.
+            bounds: Optional, parameter bounds for curve_fit.
+            fit_mean_data: Optional, if True pass means of data points to curve_fit.
+            plot: If True generate a plot of fitted data.
+            ax: Optional, matplotlib axis to add plot to.
+            kwargs: additional kwargs to pass to curve_fit.
+
+        Returns:
+            tuple: A pair ``(analysis_results, figures)`` where
+                    ``analysis_results`` may be a single or list of
+                    AnalysisResult objects, and ``figures`` may be
+                    None, a single figure, or a list of figures.
+        """
+        # Compute curve fit data
+        if fit_mean_data:
+            xdata, ydata, sigma, xraw, yraw = average_curve_fit_data(data, return_raw=True)
+        else:
+            xdata, ydata, sigma = curve_fit_data(data)
```

Comment:
This looks like a location where we could bring in the data processor. I'm not sure is we want to give `_run_curve_fit` unprocessed data and an instance of `DataProcessor` or if we give it the processed data directly.